File: frontend/lib/features/habits/data/models/requests/habit_daily_tracking.dart
class HabitDailyTrackingUpdateRequestModel {
  final DateTime datetime;
  final double quantityPerSet;
  final int quantityOfSet;
  final String unitId;
  final int weight;
  final String weightUnitId;

  const HabitDailyTrackingUpdateRequestModel({
    required this.datetime,
    required this.quantityPerSet,
    required this.quantityOfSet,
    required this.unitId,
    required this.weight,
    required this.weightUnitId,
  });

  Map<String, dynamic> toJson() {
    return {
      'datetime': datetime.toIso8601String().split('.').first,
      'quantity_per_set': quantityPerSet,
      'quantity_of_set': quantityOfSet,
      'unit_id': unitId,
      'weight': weight,
      'weight_unit_id': weightUnitId,
    };
  }
}

class HabitDailyTrackingCreateRequestModel {
  final String habitId;
  final DateTime datetime;
  final double quantityPerSet;
  final int quantityOfSet;
  final String unitId;
  final int weight;
  final String weightUnitId;
  final String? challengeDailyTracking;

  const HabitDailyTrackingCreateRequestModel({
    required this.habitId,
    required this.datetime,
    required this.quantityPerSet,
    required this.quantityOfSet,
    required this.unitId,
    required this.weight,
    required this.weightUnitId,
    required this.challengeDailyTracking,
  });

  Map<String, dynamic> toJson() {
    return {
      'habit_id': habitId,
      'datetime': datetime.toIso8601String().split('.').first,
      'quantity_per_set': quantityPerSet,
      'quantity_of_set': quantityOfSet,
      'unit_id': unitId,
      'weight': weight,
      'weight_unit_id': weightUnitId,
      'challenge_daily_tracking': challengeDailyTracking,
    };
  }
}

File: backend/src/features/habits/structs/requests/habit_daily_tracking.rs
use chrono::NaiveDateTime;

use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Deserialize)]
pub struct GetHabitDailyTrackingParams {
    pub habit_daily_tracking_id: Uuid,
}

#[derive(Deserialize)]
pub struct UpdateHabitDailyTrackingParams {
    pub habit_daily_tracking_id: Uuid,
}

#[derive(Deserialize, Serialize)]
pub struct HabitDailyTrackingUpdateRequest {
    pub datetime: NaiveDateTime,
    pub quantity_per_set: i32,
    pub quantity_of_set: i32,
    pub unit_id: Uuid,
    pub weight: i32,
    pub weight_unit_id: Uuid,
}

#[derive(Deserialize, Serialize)]
pub struct HabitDailyTrackingCreateRequest {
    pub habit_id: Uuid,
    pub datetime: NaiveDateTime,
    pub quantity_per_set: i32,
    pub quantity_of_set: i32,
    pub unit_id: Uuid,
    pub weight: i32,
    pub weight_unit_id: Uuid,
}
